### PR TITLE
Alobugdays : added "training" or "testing" param

### PR DESCRIPTION
**Pull request by Dee61298**
Solves issue #38 
Minimal reproducible example is in sintel_disparity_dataset.py in main() 

However, I wasn't able to finish the tests because the testing split is missing from /data. @ragier I heard you were the one responsible for the datas, so, should I download it again ?

### CODE REVIEW

File: alodataset/sintel_base_dataset.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ class SintelBaseDataset(BaseDataset, SequenceMixin):
         "temple_3",
     ]
 
-    def __init__(self, cameras=None, labels=None, passes=None, sintel_sequences=None, name="Sintel_base", **kwargs):
+    def __init__(self, training=True,cameras=None, labels=None, passes=None, sintel_sequences=None, name="Sintel_base", **kwargs):
```
Comment (anhtu293) [inline alodataset/sintel_base_dataset.py:58]: we support Split dataset in `alodataset`. It's cleaner to  use that for train/val split.

```
from alodataset import BaseDataset, Split, SplitMixin

class SintelBaseDataset(BaseDataset, Split, SequenceMixin):

SPLIT_FOLDERS = {Split.VAL: "validation", Split.TRAIN: "training"}
path = os.path.join(dset_dir, self.split, ...)
```

In that code `self.split` is `Split.VAL` or `Split.TRAIN`

File: alodataset/sintel_base_dataset.py [hunk 2]
```diff
@@ -67,8 +67,11 @@ def __init__(self, cameras=None, labels=None, passes=None, sintel_sequences=None
         self.sintel_sequences = sintel_sequences if sintel_sequences is not None else self.SINTEL_SEQUENCES
         if self.sample:
             return
-
         self._assert_inputs()
+        if training:
+            self.split = "training"
+        else:
+            self.split = "testing"
```
Comment (anhtu293) [inline alodataset/sintel_base_dataset.py:74]: with `Split` class, we dont need `training` argument anymore

File: alodataset/sintel_base_dataset.py [hunk 3]
```diff
@@ -98,7 +101,7 @@ def _split_cameras(self, my_dict):
 
     def _get_folder(self, sintel_seq, feature_or_label):
         dset_dir = self.dataset_dir
-        return os.path.join(dset_dir, "training", feature_or_label, sintel_seq)
+        return os.path.join(dset_dir, self.split, feature_or_label, sintel_seq)
```
Comment (anhtu293) [inline alodataset/sintel_base_dataset.py:104]: same

File: alodataset/sintel_disparity_dataset.py [hunk 4]
```diff
@@ -105,7 +105,10 @@ def _get_camera_frames(self, sequence_data, camera):
 
 
 if __name__ == "__main__":
-    dataset = SintelDisparityDataset(sample=True)
+    dataset = SintelDisparityDataset(sample=False, training=False)
```
Comment (anhtu293) [inline alodataset/sintel_disparity_dataset.py:108]: it's better to use `alodataset.Split`

File: alodataset/sintel_base_dataset.py (hunk 1 above)
Comment (Dee61298) [reply alodataset/sintel_base_dataset.py:58]: Why can't it be `Split.TEST` since that's the point of issue #38 ?

Comment (anhtu293) [reply alodataset/sintel_base_dataset.py:58]: it can be `Split.TEST` for test dataset

**Review by Data-Iab — APPROVED** [review]

**Pull request merged**